src/target/riscv/riscv-013.c

```diff
@@ -1958,11 +1958,14 @@ static int examine(struct target *target)
 	if (dm013_select_hart(target, info->index) != ERROR_OK)
 		return ERROR_FAIL;
 
-	enum riscv_hart_state state;
-	if (riscv_get_hart_state(target, &state) != ERROR_OK)
+	target->state = TARGET_UNKNOWN;
+	target->debug_reason = DBG_REASON_UNDEFINED;
```

Comment:
What does this achieve? I don't think it's accessed below, it's just unconditionally overwritten.

Reply:
corrected the message. `arp_init` -> `arp_examine`

Reply:
OK. Can you add a short comment, like `/* In case e.g. riscv013_halt_go() fails. */`

Generally I would like to see these status updates happen where we know the status changed, but that's a much larger change.

Reply:
> OK. Can you add a short comment, like /* In case e.g. riscv013_halt_go() fails. */

Addressed. Please do note that I moved these to lines way higher. I think this is more appropriate place.